### DISCUSSION TITLE
Public User Defined Types in class modules

### DISCUSSION THREAD

**Discussion by Beakerboy** [kellyethridge/VBCorLib#109] [Questions]
I have been working with Microsoft in validating their open documentation of the VBA language (MS-VBAL).

According to their documents (and support) Public UDT are not valid within class modules. However, I see you have a public type called Int64 within the Int64Static class. I wanted to double check if that construct in fact works. I’ve tried to create a class module within the VBA IDE with a public UDT, and the moment I attempt to use the object, I receive an error:

“Compile Error: Cannot define a Public user-defined type within an object module”

Is there a special trick to getting this to run without error?

**Comment by kellyethridge** [comment]
Hello,

There is no trick. It's just that this is written in VB6, which does allow Public UDTs in classes. I have no experience with VBA or any compatibility issues.

**Reply by Beakerboy** [reply]
Ah,I didn’t know that VB6 and VBA were different. Thanks for the quick response! I’m writing a VBA parsing and Linting tool and someone suggested I run it against your code to find bugs in my tool. it originally spat out an error for this,  but I was able to import and export VBA with public UDTs to and from the VBA IDE, so I submitted a documentation bug. They confirmed that VBA will not flag a problem until runtime so they said they would add a note to the documentation about that.

Thanks again.
